corehq/apps/userreports/views.py

```diff
@@ -439,32 +443,97 @@ def preview_data_source(request, domain, config_id):
     return render(request, "userreports/preview_data.html", context)
 
 
+ExportParameters = namedtuple('ExportParameters',
+                              ['format', 'keyword_filters', 'sql_filters'])
+
+
+def _last_n_days(column, value):
+    if not isinstance(column.type, (types.Date, types.DateTime)):
+        raise UserQueryError(_("You can only use 'lastndays' on date columns"))
+    end = datetime.date.today()
+    start = end - datetime.timedelta(days=int(value))
+    return column.between(start, end)
+
+
+def _range_filter(column, value):
+    try:
+        start, end = value.split('..')
+    except ValueError:
+        raise UserQueryError(_('Ranges must have the format "start..end"'))
+    return column.between(start, end)
+
+
+sql_directives = [
+    # (suffix matching url parameter, callable returning a filter),
+    ('-lastndays', _last_n_days),
+    ('-range', _range_filter),
+]
+
+
+def process_url_params(params, columns):
+    """
+    Converts a dictionary of parameters from the user to sql filters.
+
+    If a parameter is of the form <field name>-<suffix>, where suffix is
+    defined in `sql_directives`, the corresponding function is used to
+    produce a filter.
+    """
+    format_ = params.get('$format', Format.UNZIPPED_CSV)
```

Comment:
yeah, just a way to be like "format, but not THAT format".  It's a way to not override builtins.  (That said, if this is more confusing than the alternative, I've got no problem just using `format`)